cnd/src/main.rs

```diff
@@ -160,14 +165,19 @@ fn version() {
     println!("{} {} ({})", name, version, short);
 }
 
+fn port_is_available(port: u16) -> bool {
+    match TcpListener::bind(("127.0.0.1", port)) {
+        Ok(_) => true,
+        Err(_) => false,
+    }
+}
+
 async fn spawn_warp_instance(settings: Settings, dependencies: Facade) {
     let routes = route_factory::create(dependencies, &settings.http_api.cors.allowed_origins);
-
     let listen_addr = settings.http_api.socket;
 
-    tracing::info!("Starting HTTP server on {}", listen_addr);
-
-    warp::serve(routes).bind(listen_addr).await
+    tracing::info!("Starting HTTP server on {} ...", listen_addr);
+    warp::serve(routes).try_bind(listen_addr).await
```

Comment:
```suggestion
    warp::serve(routes).bind(listen_addr).await
```
`try_bind` returns a `Result`. I think we should keep `bind` so it panics.

Reply:
We can return a result from `main` and avoid the panic by just shutting down the application like in any other error.
That would give us a nice error backtrace through anyhow instead of panicing.

I am in favor of `try_bind`.

Reply:
Just make sure to handle the result then an bubble the error up in case it fails.

Reply:
`try_bind` does not return a result.

Reply:
``` warp server.rs

    /// Bind to a socket address, returning a `Future` that can be
    /// executed on any runtime.
    ///
    /// In case we are unable to bind to the specified address, resolves to an
    /// error and logs the reason.
    pub async fn try_bind(self, addr: impl Into<SocketAddr> + 'static) {
        let addr = addr.into();
        let srv = match try_bind!(self, &addr) {
            Ok((_, srv)) => srv,
            Err(err) => {
                log::error!("error binding to {}: {}", addr, err);
                return;
            }
        };

        srv.map(|result| {
            if let Err(err) = result {
                log::error!("server error: {}", err)
            }
        })
        .await;
    }
```

Reply:
Lol, why would it not return an error :roll_eyes:

Reply:
Stupid eh, especially confusing given the name.